client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java

```diff
@@ -231,12 +231,14 @@ public ConfigurationBuilder keySizeEstimate(int keySizeEstimate) {
    @Override
    public ConfigurationBuilder marshaller(String marshaller) {
       this.marshallerClass = Util.loadClass(marshaller, this.classLoader());
+      this.marshaller = marshallerClass == null ? null : Util.getInstance(marshallerClass);
```

Comment:
This is still wrong. ```marshallerClass == null``` will never happen here because the line above would have failed already. Util.loadClass never returns null. 

I propose removing a bit of confusion  by renaming parameter ```String marshaller``` to ```String marshallerClassName```.  And avoid referencing a field both with and without ````this.```